### PR TITLE
feat(logging): export redactLogLine function

### PR DESCRIPTION
**Pull request by Charge71**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/mean-ants-hang.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/backend-common': minor
```
Comment (benjdlambert) [inline .changeset/mean-ants-hang.md:2]:
```suggestion
'@backstage/backend-common': patch
```

File: .changeset/mean-ants-hang.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Exported redactLogLine function to be able to use it in custom loggers.
```
Comment (benjdlambert) [inline .changeset/mean-ants-hang.md:5]:
```suggestion
Exported `redactLogLine` function to be able to use it in custom loggers.
```

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Just some small things in the changeset.

**Comment by benjdlambert** [comment]
@Charge71 sorry, looks like the suggestions don't automatically sign off the commits 😅 looks like you'll have to rebase this and signoff the last two commits.

File: packages/backend-common/api-report.md [hunk 3]
```diff
@@ -568,6 +568,11 @@ export type ReadUrlResponseFactoryFromStreamOptions = {
   etag?: string;
 };
 
+// @public
+export function redactLogLine(
```
Comment (benjdlambert) [inline packages/backend-common/api-report.md:572]: Also, one more thing that I've just thought about, maybe we should call this `redactWinstonLogLine` or something instead to namespace it a little better.

**Review by benjdlambert — APPROVED** [review]
OK - let's go with this. It's a little tricky as I really don't think we should be exporting this, but there's not really a better solution I don't think. Merging with the caveat that it might change in upcoming version and should be treat as unstable.

**Comment by Charge71** [comment]
Thank you. I agree is not the best solution but we couldn't find a better way to access the secrets from our custom logger.

**Pull request merged**